chips/stm32f4xx/src/usart.rs

```diff
@@ -12,7 +12,7 @@ use crate::rcc;
 
 /// Universal synchronous asynchronous receiver transmitter
 #[repr(C)]
-struct UsartRegisters {
+pub struct UsartRegisters {
```

Comment:
Is it ok (from a safety perspective) to make register structs public?

Reply:
This is already done in a  number of other places throughout the `chips` and `arch` directories. We are protected against capsules attempting to use something like this directly by the fact that capsule code depends only on the core kernel, not on anything in `chips/` or `arch/`.

Reply:
Also these register structs are only "unsafe" in the sense that you are creating them over some memory address using `StaticRef`. Without resorting to `unsafe` in the first place, you shouldn't be able to write unsound code with those structs (as they would then operate over memory they own).

Reply:
Ok. Makes sense.